string model_file_location
int num_end_effectors
string[] end_effector_names